src/SourceBuild/content/prep-source-build.sh

```diff
@@ -223,34 +223,35 @@ fi
 
 if [ "$removeBinaries" == true ]; then
 
+  originalPackagesDir=$packagesDir
+  # Create working directory for extracking packages
+  workingDir=$(mktemp -d)
+
   # If --with-packages is not passed, unpack PSB artifacts
   if [[ $packagesDir == $defaultPackagesDir ]]; then
+    echo "  Extracting previously source-built to $workingDir"
     sourceBuiltArchive=$(find "$packagesArchiveDir" -maxdepth 1 -name 'Private.SourceBuilt.Artifacts*.tar.gz')
 
-    if [ ! -d "$packagesDir" ] && [ -f "$sourceBuiltArchive" ]; then
-      echo "  Unpacking Private.SourceBuilt.Artifacts.*.tar.gz into $packagesDir"
-      mkdir -p "$packagesDir"
-      tar -xzf "$sourceBuiltArchive" -C "$packagesDir"
-    elif [ -f "$sourceBuiltArchive" ]; then
-      if [ ! -f "$packagesDir/PackageVersions.props" ]; then
-        echo "  Creating $packagesDir/PackageVersions.props..."
-        tar -xzf "$sourceBuiltArchive" -C "$packagesDir" PackageVersions.props
-      fi
-      if [ ! -f "$packagesDir/VerticalManifest.xml" ]; then
-        echo "  Unpacking Asset manifests into $packagesDir..."
-        tar -xzf "$sourceBuiltArchive" -C "$packagesDir" VerticalManifest.xml
-      fi
-    else
+    if [ ! -f "$sourceBuiltArchive" ]; then
```

Comment:
@jkoritzinsky does this change look okay to you? Essentially, we always extract the `$sourceBuiltArchive` to a new temporary directory and so don't check if the PackageVersions.props or VerticalManifest exist.

Reply:
It looks good to me.